subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/EclipseModel.java

```diff
@@ -226,14 +231,12 @@ public void jdt(Action<? super EclipseJdt> action) {
     /**
      * The list of tasks to execute before the Eclipse synchronization starts.
      *
-     *
-     *
      * @return the tasks names
      * @since 5.4
      */
     @Incubating
-    public List<String> getSynchronizationTasks() {
-        return this.synchronizationTasks;
+    public DefaultTaskDependency getSynchronizationTasks() {
```

Comment:
Should not expose internal type (just return an immutable set)

Reply:
Shoot, I forgot to add a todo there :) Fixed.